### PR TITLE
resource field updates

### PR DESCRIPTION
**Pull request by dlsteuer**
- allow the name field for customers
- add trial_period_days field
- when on a free trial don't invoice (specified by trial_end or trial_period_days)

### CODE REVIEW

File: localstripe/resources.py [hunk 1]
```diff
@@ -558,6 +558,7 @@ def __init__(self, description=None, email=None, invoice_settings=None,
         self.discount = None
         self.shipping = None
         self.default_source = None
+        self.name = name or ''
```
Comment (adrienverge) [inline localstripe/resources.py]: If `name` is provided, it needs to be checked to be a string (like for `email` and `description` above, line 518).

Also, can you put it before `email` and `description`, for clarity? (both here and in `__init__()` arguments)

File: localstripe/resources.py [hunk 2]
```diff
@@ -2058,12 +2060,15 @@ def __init__(self, customer=None, metadata=None, items=None,
         self.status = 'incomplete'
         self.trial_end = None
         self.trial_start = None
+        self.trial_period_days = trial_period_days
```
Comment (adrienverge) [inline localstripe/resources.py]: If `trial_period_days` is provided, it needs to be checked to be an integer. See an example here for `Charge`:
https://github.com/adrienverge/localstripe/blob/3d0110ac0df320b24ffe275a5ba3645a97eef2e6/localstripe/resources.py#L328

**Comment by dlsteuer** [comment]
Sure thing, changes have been, I think I got the validation the way you asked for.

**Comment by adrienverge** [comment]
I've split your PR into two commits for clarity:

- feat(customers): add support for names in customers
- feat(subscriptions): don't invoice when on a free trial

**Pull request merged**